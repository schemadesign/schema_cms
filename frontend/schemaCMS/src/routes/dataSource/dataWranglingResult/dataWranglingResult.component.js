import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { path, pathEq } from 'ramda';
import { FormattedMessage } from 'react-intl';

import messages from './dataWranglingResult.messages';
import DataPreview from '../../../shared/components/dataPreview/dataPreview.component';
import { DataSourceNavigation } from '../../../shared/components/dataSourceNavigation';
import { TopHeader } from '../../../shared/components/topHeader';
import { ContextHeader } from '../../../shared/components/contextHeader';
import { SOURCE_PAGE } from '../../../modules/dataSource/dataSource.constants';

export class DataWranglingResult extends PureComponent {
  static propTypes = {
    previewData: PropTypes.object.isRequired,
    dataSource: PropTypes.object,
    fetchPreview: PropTypes.func.isRequired,
    intl: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }),
  };

  render() {
    const { dataSource } = this.props;
    const activeJobId = path(['activeJob', 'id'], dataSource);
    const headerTitle = dataSource.name;
    const headerSubtitle = <FormattedMessage {...messages.subTitle} />;
    const isFakeJob = pathEq(['activeJob', 'scripts'], [], dataSource);

    if (isFakeJob) {
      this.props.history.push(`/datasource/${dataSource.id}/${SOURCE_PAGE}`);
      return null;
    }

    return (
      <Fragment>
        <Helmet title={this.props.intl.formatMessage(messages.pageTitle)} />
        <TopHeader headerTitle={headerTitle} headerSubtitle={headerSubtitle} />
        <ContextHeader title={headerTitle} subtitle={headerSubtitle}>
          <DataSourceNavigation {...this.props} />
        </ContextHeader>
        <DataPreview {...this.props} jobId={activeJobId} />
        <DataSourceNavigation {...this.props} hideOnDesktop />
      </Fragment>
    );
  }
}
