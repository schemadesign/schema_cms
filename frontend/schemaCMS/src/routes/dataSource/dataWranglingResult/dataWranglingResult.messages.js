/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'dataSource.dataWranglingResult.pageTitle',
    defaultMessage: 'Data Wrangling Result',
  },
  subTitle: {
    id: 'dataSource.dataWranglingResult.pageTitle',
    defaultMessage: 'Data Wrangling Result',
  },
});
