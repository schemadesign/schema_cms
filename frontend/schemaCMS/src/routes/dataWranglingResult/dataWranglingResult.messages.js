/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'dataWranglingResult.pageTitle',
    defaultMessage: 'Data Wrangling Result',
  },
});
