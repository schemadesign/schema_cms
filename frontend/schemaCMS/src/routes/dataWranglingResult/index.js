export { default as DataWranglingResult } from './dataWranglingResult.container';
