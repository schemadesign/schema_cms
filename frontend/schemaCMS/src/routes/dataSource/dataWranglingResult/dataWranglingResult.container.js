import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { DataWranglingResult } from './dataWranglingResult.component';
import { JobRoutines, selectJobPreview } from '../../../modules/job';
import { selectDataSource } from '../../../modules/dataSource';
import { selectUserRole } from '../../../modules/userProfile';

const mapStateToProps = createStructuredSelector({
  userRole: selectUserRole,
  previewData: selectJobPreview,
  dataSource: selectDataSource,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      fetchPreview: promisifyRoutine(JobRoutines.fetchPreview),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(mapStateToProps, mapDispatchToProps),
  injectIntl,
  withRouter
)(DataWranglingResult);
