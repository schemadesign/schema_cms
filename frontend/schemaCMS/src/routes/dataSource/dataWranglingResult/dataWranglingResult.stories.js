import React from 'react';
import { storiesOf } from '@storybook/react';

import { withTheme } from '../../../.storybook/decorators';
import { history, intl } from '../../../.storybook/helpers';
import { DataWranglingResult } from './dataWranglingResult.component';
import { tableData as data, tableFields as fields } from '../../../shared/utils/dataMock';
import { JOB_STATE_FAILURE, JOB_STATE_SUCCESS } from '../../../modules/job/job.constants';

export const defaultProps = {
  dataSource: {
    jobs: [{ id: 1, jobState: JOB_STATE_FAILURE }, { id: 2, jobState: JOB_STATE_SUCCESS }],
    metaData: {},
    project: { id: '1' },
  },
  previewData: {
    fields,
    data,
  },
  fetchPreview: Function.prototype,
  history,
  intl,
  match: {
    params: {
      projectId: '1',
      dataSourceId: '1',
      step: '4',
    },
    url: 'url',
  },
};

storiesOf('Data Source|DataWranglingResult', module)
  .addDecorator(withTheme())
  .add('Default', () => <DataWranglingResult {...defaultProps} />);
