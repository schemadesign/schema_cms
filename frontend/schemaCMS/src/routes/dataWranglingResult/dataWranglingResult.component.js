import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { path } from 'ramda';

import messages from './dataWranglingResult.messages';
import { Container } from './dataWranglingResult.styles';
import { StepNavigation } from '../../shared/components/stepNavigation';
import DataPreview from '../../shared/components/dataPreview/dataPreview.component';

export class DataWranglingResult extends PureComponent {
  static propTypes = {
    previewData: PropTypes.object.isRequired,
    dataSource: PropTypes.object,
    fetchPreview: PropTypes.func.isRequired,
    intl: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }),
  };

  getActiveJobId = () => path(['dataSource', 'activeJob', 'id'], this.props);

  render() {
    const activeJobId = this.getActiveJobId(this.props);

    return (
      <Container>
        <Helmet title={this.props.intl.formatMessage(messages.pageTitle)} />
        <DataPreview {...this.props} jobId={activeJobId} />
        <StepNavigation {...this.props} />
      </Container>
    );
  }
}
