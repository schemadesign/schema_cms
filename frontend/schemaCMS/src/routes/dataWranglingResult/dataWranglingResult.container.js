import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';

import { DataWranglingResultRoutines, selectFields, selectPreviewTable } from '../../modules/dataWranglingResult';
import { DataWranglingResult } from './dataWranglingResult.component';

const mapStateToProps = createStructuredSelector({
  fields: selectFields,
  previewTable: selectPreviewTable,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      fetchResult: promisifyRoutine(DataWranglingResultRoutines.fetch),
      unmountResult: promisifyRoutine(DataWranglingResultRoutines.unmount),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(DataWranglingResult);
