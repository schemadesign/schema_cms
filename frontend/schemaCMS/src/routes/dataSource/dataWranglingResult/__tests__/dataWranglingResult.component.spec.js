import React from 'react';
import { shallow } from 'enzyme';

import { DataWranglingResult } from '../dataWranglingResult.component';
import { defaultProps } from '../dataWranglingResult.stories';

describe('DataWranglingResult: Component', () => {
  const component = props => <DataWranglingResult {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly with loading', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', async () => {
    defaultProps.fetchPreview = jest.fn().mockReturnValue(Promise.resolve());
    const wrapper = await render(defaultProps);

    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect on no job', async () => {
    jest.spyOn(defaultProps.history, 'push');

    render({
      dataSource: {
        ...defaultProps.dataSource,
        activeJob: null,
      },
    });

    expect(defaultProps.history.push).toBeCalledWith('/datasource/1/source');
  });
});
