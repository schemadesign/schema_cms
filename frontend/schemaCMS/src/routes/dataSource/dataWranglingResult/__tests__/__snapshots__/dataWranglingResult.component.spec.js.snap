// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataWranglingResult: Component should render correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Wrangling Result"
  />
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Data Wrangling Result"
        id="dataWranglingResult.pageTitle"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Data Wrangling Result"
        id="dataWranglingResult.pageTitle"
        values={Object {}}
      />
    }
  >
    <DataSourceNavigation
      dataSource={
        Object {
          "jobs": Array [
            Object {
              "id": 1,
              "jobState": "failed",
            },
            Object {
              "id": 2,
              "jobState": "success",
            },
          ],
          "metaData": Object {},
        }
      }
      fetchPreview={[Function]}
      fetchResult={[MockFunction]}
      hideOnDesktop={false}
      history={
        Object {
          "push": [Function],
        }
      }
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "dataSourceId": "1",
            "projectId": "1",
            "step": "4",
          },
        }
      }
      previewData={
        Object {
          "data": Array [
            Object {
              "Birthday": "02/22/1732",
              "Birthplace": "Westmoreland County",
              "First Name": "George",
              "Full Name": "George Washington",
              "Last Name": "Washington",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "10/30/1735",
              "Birthplace": "Braintree",
              "First Name": "John",
              "Full Name": "John Adams",
              "Last Name": "Adams",
              "State of Birth": "MA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "04/13/1743",
              "Birthplace": "Shadwell",
              "First Name": "Thomas",
              "Full Name": "Thomas Jefferson",
              "Last Name": "Jefferson",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "03/16/1751",
              "Birthplace": "Port Conway",
              "First Name": "James",
              "Full Name": "James Madison",
              "Last Name": "Madison",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "04/28/1758",
              "Birthplace": "Monroe Hall",
              "First Name": "James",
              "Full Name": "James Monroe",
              "Last Name": "Monroe",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
          ],
          "fields": Object {
            "Birthday": Object {},
            "Birthplace": Object {},
            "First Name": Object {},
            "Full Name": Object {},
            "Last Name": Object {},
            "State of Birth": Object {},
            "Url": Object {},
          },
        }
      }
    />
  </ContextHeader>
  <DataPreview
    dataSource={
      Object {
        "jobs": Array [
          Object {
            "id": 1,
            "jobState": "failed",
          },
          Object {
            "id": 2,
            "jobState": "success",
          },
        ],
        "metaData": Object {},
      }
    }
    fetchPreview={[Function]}
    fetchResult={[MockFunction]}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    jobId={null}
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "4",
        },
      }
    }
    previewData={
      Object {
        "data": Array [
          Object {
            "Birthday": "02/22/1732",
            "Birthplace": "Westmoreland County",
            "First Name": "George",
            "Full Name": "George Washington",
            "Last Name": "Washington",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "10/30/1735",
            "Birthplace": "Braintree",
            "First Name": "John",
            "Full Name": "John Adams",
            "Last Name": "Adams",
            "State of Birth": "MA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/13/1743",
            "Birthplace": "Shadwell",
            "First Name": "Thomas",
            "Full Name": "Thomas Jefferson",
            "Last Name": "Jefferson",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "03/16/1751",
            "Birthplace": "Port Conway",
            "First Name": "James",
            "Full Name": "James Madison",
            "Last Name": "Madison",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/28/1758",
            "Birthplace": "Monroe Hall",
            "First Name": "James",
            "Full Name": "James Monroe",
            "Last Name": "Monroe",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
        ],
        "fields": Object {
          "Birthday": Object {},
          "Birthplace": Object {},
          "First Name": Object {},
          "Full Name": Object {},
          "Last Name": Object {},
          "State of Birth": Object {},
          "Url": Object {},
        },
      }
    }
  />
  <DataSourceNavigation
    dataSource={
      Object {
        "jobs": Array [
          Object {
            "id": 1,
            "jobState": "failed",
          },
          Object {
            "id": 2,
            "jobState": "success",
          },
        ],
        "metaData": Object {},
      }
    }
    fetchPreview={[Function]}
    fetchResult={[MockFunction]}
    hideOnDesktop={true}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "4",
        },
      }
    }
    previewData={
      Object {
        "data": Array [
          Object {
            "Birthday": "02/22/1732",
            "Birthplace": "Westmoreland County",
            "First Name": "George",
            "Full Name": "George Washington",
            "Last Name": "Washington",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "10/30/1735",
            "Birthplace": "Braintree",
            "First Name": "John",
            "Full Name": "John Adams",
            "Last Name": "Adams",
            "State of Birth": "MA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/13/1743",
            "Birthplace": "Shadwell",
            "First Name": "Thomas",
            "Full Name": "Thomas Jefferson",
            "Last Name": "Jefferson",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "03/16/1751",
            "Birthplace": "Port Conway",
            "First Name": "James",
            "Full Name": "James Madison",
            "Last Name": "Madison",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/28/1758",
            "Birthplace": "Monroe Hall",
            "First Name": "James",
            "Full Name": "James Monroe",
            "Last Name": "Monroe",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
        ],
        "fields": Object {
          "Birthday": Object {},
          "Birthplace": Object {},
          "First Name": Object {},
          "Full Name": Object {},
          "Last Name": Object {},
          "State of Birth": Object {},
          "Url": Object {},
        },
      }
    }
  />
</Fragment>
`;

exports[`DataWranglingResult: Component should render correctly with loading 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Wrangling Result"
  />
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Data Wrangling Result"
        id="dataWranglingResult.pageTitle"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Data Wrangling Result"
        id="dataWranglingResult.pageTitle"
        values={Object {}}
      />
    }
  >
    <DataSourceNavigation
      dataSource={
        Object {
          "jobs": Array [
            Object {
              "id": 1,
              "jobState": "failed",
            },
            Object {
              "id": 2,
              "jobState": "success",
            },
          ],
          "metaData": Object {},
        }
      }
      fetchPreview={[Function]}
      hideOnDesktop={false}
      history={
        Object {
          "push": [Function],
        }
      }
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "dataSourceId": "1",
            "projectId": "1",
            "step": "4",
          },
        }
      }
      previewData={
        Object {
          "data": Array [
            Object {
              "Birthday": "02/22/1732",
              "Birthplace": "Westmoreland County",
              "First Name": "George",
              "Full Name": "George Washington",
              "Last Name": "Washington",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "10/30/1735",
              "Birthplace": "Braintree",
              "First Name": "John",
              "Full Name": "John Adams",
              "Last Name": "Adams",
              "State of Birth": "MA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "04/13/1743",
              "Birthplace": "Shadwell",
              "First Name": "Thomas",
              "Full Name": "Thomas Jefferson",
              "Last Name": "Jefferson",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "03/16/1751",
              "Birthplace": "Port Conway",
              "First Name": "James",
              "Full Name": "James Madison",
              "Last Name": "Madison",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "04/28/1758",
              "Birthplace": "Monroe Hall",
              "First Name": "James",
              "Full Name": "James Monroe",
              "Last Name": "Monroe",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
          ],
          "fields": Object {
            "Birthday": Object {},
            "Birthplace": Object {},
            "First Name": Object {},
            "Full Name": Object {},
            "Last Name": Object {},
            "State of Birth": Object {},
            "Url": Object {},
          },
        }
      }
    />
  </ContextHeader>
  <DataPreview
    dataSource={
      Object {
        "jobs": Array [
          Object {
            "id": 1,
            "jobState": "failed",
          },
          Object {
            "id": 2,
            "jobState": "success",
          },
        ],
        "metaData": Object {},
      }
    }
    fetchPreview={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    jobId={null}
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "4",
        },
      }
    }
    previewData={
      Object {
        "data": Array [
          Object {
            "Birthday": "02/22/1732",
            "Birthplace": "Westmoreland County",
            "First Name": "George",
            "Full Name": "George Washington",
            "Last Name": "Washington",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "10/30/1735",
            "Birthplace": "Braintree",
            "First Name": "John",
            "Full Name": "John Adams",
            "Last Name": "Adams",
            "State of Birth": "MA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/13/1743",
            "Birthplace": "Shadwell",
            "First Name": "Thomas",
            "Full Name": "Thomas Jefferson",
            "Last Name": "Jefferson",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "03/16/1751",
            "Birthplace": "Port Conway",
            "First Name": "James",
            "Full Name": "James Madison",
            "Last Name": "Madison",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/28/1758",
            "Birthplace": "Monroe Hall",
            "First Name": "James",
            "Full Name": "James Monroe",
            "Last Name": "Monroe",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
        ],
        "fields": Object {
          "Birthday": Object {},
          "Birthplace": Object {},
          "First Name": Object {},
          "Full Name": Object {},
          "Last Name": Object {},
          "State of Birth": Object {},
          "Url": Object {},
        },
      }
    }
  />
  <DataSourceNavigation
    dataSource={
      Object {
        "jobs": Array [
          Object {
            "id": 1,
            "jobState": "failed",
          },
          Object {
            "id": 2,
            "jobState": "success",
          },
        ],
        "metaData": Object {},
      }
    }
    fetchPreview={[Function]}
    hideOnDesktop={true}
    history={
      Object {
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
          "projectId": "1",
          "step": "4",
        },
      }
    }
    previewData={
      Object {
        "data": Array [
          Object {
            "Birthday": "02/22/1732",
            "Birthplace": "Westmoreland County",
            "First Name": "George",
            "Full Name": "George Washington",
            "Last Name": "Washington",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "10/30/1735",
            "Birthplace": "Braintree",
            "First Name": "John",
            "Full Name": "John Adams",
            "Last Name": "Adams",
            "State of Birth": "MA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/13/1743",
            "Birthplace": "Shadwell",
            "First Name": "Thomas",
            "Full Name": "Thomas Jefferson",
            "Last Name": "Jefferson",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "03/16/1751",
            "Birthplace": "Port Conway",
            "First Name": "James",
            "Full Name": "James Madison",
            "Last Name": "Madison",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/28/1758",
            "Birthplace": "Monroe Hall",
            "First Name": "James",
            "Full Name": "James Monroe",
            "Last Name": "Monroe",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
        ],
        "fields": Object {
          "Birthday": Object {},
          "Birthplace": Object {},
          "First Name": Object {},
          "Full Name": Object {},
          "Last Name": Object {},
          "State of Birth": Object {},
          "Url": Object {},
        },
      }
    }
  />
</Fragment>
`;
